File: Mobile development/Lab8/app/src/main/java/iu9/bmstu/ru/lab8/MainActivity.java
package iu9.bmstu.ru.lab8;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.Snackbar;
import android.support.v4.app.ActivityCompat;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private static final int PERMISSION_REQUEST_CODE = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button buttonFileManager = findViewById(R.id.button_open_filemanager);
        buttonFileManager.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (ContextCompat.checkSelfPermission(getApplicationContext(),
                        Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                        ActivityCompat.requestPermissions(MainActivity.this,
                                new String[]{Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE},
                                PERMISSION_REQUEST_CODE);
                    }
                }
                else {
                    startActivity(new Intent(MainActivity.this,FileActivity.class));
                }
            }
        });
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        switch (requestCode) {
            case PERMISSION_REQUEST_CODE: {
                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                    startActivity(new Intent(this,FileActivity.class));
                } else {
                    Snackbar.make(findViewById(R.id.main_activity), "Нет разрешения для чтения/записи внешней памяти", Snackbar.LENGTH_LONG).show();
                    ActivityCompat.requestPermissions(MainActivity.this,
                            new String[]{Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE},
                            PERMISSION_REQUEST_CODE);
                }
                break;
            }
        }

    }

}

File: Mobile development/Lab1/MyApplication/app/src/main/java/iu9/bmstu/ru/lab1/MainActivity.kt
package iu9.bmstu.ru.lab1

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import android.widget.NumberPicker
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private var textView: TextView? = null
    private var editText: EditText? = null
    private var imageView: ImageView? = null
    private var numberPicker: NumberPicker? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textView = findViewById(R.id.textView) as TextView
        editText = findViewById(R.id.editText) as EditText
        imageView = findViewById(R.id.imageView) as ImageView
        numberPicker = findViewById(R.id.numberPicker) as NumberPicker
        numberPicker?.maxValue = 9
        numberPicker?.minValue = 0

        textView?.text = "new text"

        textView?.setOnClickListener { Toast.makeText(baseContext, "Toast!", Toast.LENGTH_LONG).show() }
    }
}
